pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ConsistentHashingStickyKeyConsumerSelector.java

```diff
@@ -109,6 +116,11 @@ public void removeConsumer(Consumer consumer) {
                     }
                 }
             }
+            ConsumerHashAssignmentsSnapshot assignmentsAfter = internalGetConsumerHashAssignmentsSnapshot();
+            ImpactedConsumersResult impactedConsumers =
+                    consumerHashAssignmentsSnapshot.resolveImpactedConsumers(assignmentsAfter);
```

Comment:
It's not a blocker, we can figure if it's make sense to avoid taking the whole snapshot when for getting impacted consumers.

Reply:
What is your concern about the snapshot? Is it memory or CPU consumption?
I have optimized the snapshotting algorithm. In profiling, I noticed that the biggest bottleneck was adding entries to a TreeMap. Since the ranges are already sorted, that's unnecessary. The optimized implementation performs adding of 1000 consumers with 100 hash ring points in about 1 second on Apple M3. It's snapshotting after every change.

Reply:
> Is there any reason to take a whole snapshot for getting the impacted consumers when adding or removing a consumer? 

The main purpose is to have a reliable way to detect which hash ranges got removed from existing consumers.

Reply:
I have no major concern. Just curious about why not getting the impacted consumers directly when putting or removing the consumer from the Hash Ring.

Reply:
> I have no major concern. Just curious about why not getting the impacted consumers directly when putting or removing the consumer from the Hash Ring.

Yes, that would be interesting to check later if there's a way to make it faster. The main benefit of the current solution is that it's relatively easy to ensure the correctness of it and apply the solution to any selector implementation.

Reply:
Here's a follow up to PIP-379 regarding the performance concerns of snapshotting:
https://github.com/apache/pulsar/pull/23423
The snapshots are only needed for AUTO_SPLIT ordered mode. The changes in the PR will make it so that all other use cases don't use snapshotting.
Any unnecessary work that can be avoided is a useful optimization, although the performance of the current snapshotting isn't a concern based on the microbenchmark.